src/representation_theory/fdRep.lean

```diff
@@ -115,26 +129,26 @@ namespace fdRep
 
 open representation
 
```

Comment:
I would really like to be able to write here

```
open representation

def ihom (X : fdRep k G) : fdRep k G ⥤ fdRep k G :=
{ obj := λ Y, fdRep.of (lin_hom X.ρ Y.ρ),
  map := λ Y Y' f, sorry, }

def ihom_adjunction (X : fdRep k G) : monoidal_category.tensor_left X ⊣ ihom X := sorry

instance (X : fdRep k G) : closed X :=
{ is_adj :=
  { right := ihom X,
    adj := ihom_adjunction X, }, }

instance : monoidal_closed (fdRep k G) :=
{ closed' := λ X, by apply_instance, }
```

and then be able to write `V ⟶[fdRep k G] W` in place of `fdRep.of (lin_hom ρV ρW)`.

But I guess this PR shouldn't wait for it. :-(

Reply:
(In particular this would hopefully avoid the `rw [show ...]` business in the orthogonality proof.)

Reply:
Oh dear, actually just
```
def foo (V W : fdRep k G) : (Vᘁ ⊗ W) ≅ (Vᘁ ⊗ W) :=
sorry
```
immediately after the `right_rigid_category` instance causes a timeout. Something is definitely screwed up there.

Reply:
(This comment thread is not intended to block this PR in any way; I'd like to hear if you have any thoughts on these problems, though!)

Reply:
I agree that eventually the use of `representation.lin_hom` and `representation.dual` should be replaced by the monoidal closed and rigid structures on `fdRep k G` eventually. I've worked a little bit on that but it's quite difficult; we now almost have the monoidal closed structure on `Action V G`  (it's the transfer of [category_theory.functor.monoidal_closed](https://leanprover-community.github.io/mathlib_docs/category_theory/closed/functor_category.html#category_theory.functor.monoidal_closed) by [Action.functor_category_monoidal_equivalence](https://leanprover-community.github.io/mathlib_docs/representation_theory/Action.html#Action.functor_category_monoidal_equivalence)) but proving that it corresponds to the concrete description of `representation.lin_hom` is tricky. There are similar difficulties for the rigid structure, in fact I think it's worse because our API for rigid categories is very limited.
I've also encountered similar weird timeouts before when working on that stuff but I couldn't figure out what was going on.